File: ocel/examples/import_xmlocel_export_jsonocel.py
import ocel


def execute_script():
    print("validated input", ocel.validate("../logs/minimal.xmlocel", "../schemas/schema.xml"))
    log = ocel.import_log("../logs/minimal.xmlocel")
    ocel.export_log(log, "log2.jsonocel")
    print("validated output", ocel.validate("log2.jsonocel", "../schemas/schema.json"))
    ocel.import_log("log2.jsonocel")


if __name__ == "__main__":
    execute_script()

File: onprom-ocel/src/main/java/it/unibz/inf/pm/ocel/examples/ImportXmlocelExportJsonocel.java
package it.unibz.inf.pm.ocel.examples;

import it.unibz.inf.pm.ocel.OcelInitial;
import org.dom4j.DocumentException;

import java.util.Map;

public class ImportXmlocelExportJsonocel {

    public static void main(String[] args) throws Exception {
        try {
            System.out.println("validated input:");
            OcelInitial.validate("ocel/logs/minimal.xmlocel", "ocel/schemas/schema.xml");
            Map log = (Map)OcelInitial.import_log("ocel/logs/minimal.xmlocel");
            OcelInitial.export_log(log,"ocel/examples/log2.jsonocel");

            System.out.println("\nvalidated output:");
            OcelInitial.validate("ocel/examples/log2.jsonocel", "ocel/schemas/schema.json");
        } catch (DocumentException e) {
            e.printStackTrace();
        }
    }
}
